lib/solveEquation/stepThrough.js

```diff
@@ -182,12 +295,35 @@ function step(equation, symbolName) {
 function addSimplificationSteps(steps, equation, debug=false) {
   let oldEquation = equation.clone();
 
+  /*
+    Collects substeps for factoring and simplifying the expression separately
+    because their logics are located in different places. Collect factoring
+    substeps first in case the original expression is factorable. If no factoring
+    substeps, proceed to collect simplifying substeps.
+    Some examples of when this comes into play:
+    e.g. x * x + 3x + 2 = 0 <- factor the left side
+    e.g. x + 4 + 2 = 0 <- simplify the left side
+  */
   const leftSteps = simplifyExpressionNode(equation.leftNode, false);
+  const simplifiedLeftNode = leftSteps.length !== 0
+        ? leftSteps.slice(-1)[0].newNode
+        : equation.leftNode;
+  const leftFactorSteps = factor(simplifiedLeftNode, false);
+
   const leftSubSteps = [];
-  for (let i = 0; i < leftSteps.length; i++) {
-    const step = leftSteps[i];
+
+  for (let i = 0; i < leftFactorSteps.length; i++) {
+    const step = leftFactorSteps[i];
     leftSubSteps.push(EquationStatus.addLeftStep(equation, step));
   }
+
+  if (leftSubSteps.length === 0) {
```

Comment:
How am I mixing left simplify steps and right factor steps? I'm confused on that part.